web/app/js/components/TapQueryForm.jsx

```diff
@@ -183,14 +181,14 @@ class TapQueryForm extends React.Component {
 
       this.setState(state);
       this.props.updateQuery(state.query);
+      this.handleUrlUpdate(state.query);
```

Comment:
For my own understanding, can you explain why you moved `this.handleUrlUpdate()` out of the `if (!_isNil(scopeResource)) {` block that starts on line 172? The previous code only ran `handleUrlUpdate()` if the `scopeResource` is not nil. This branch is running it in all cases.

Reply:
Due to `setQuery` accepts more than one value and replace them in the current `query` prop, we don't need to call `handleUrlUpdate()` every time that some value changes. For that, we are going to call it only once with all values. Although, we only need to pass `formVal` and `scopeResource` (if not nil) values to `handleUrlUpdate()`. If you prefer this, I can change it.

Reply:
Thanks for the explanation -- I think passing `formVal` and `scopeResource` to `handleUrlUpdate()` instead of the entire `state.query` makes it easier for others to figure out what's going on in that function, so I agree with making that change 👍